### ISSUE TITLE
Issues navigating away from map with multiple base layers

### ISSUE THREAD

**Issue by dkrichards86** [PaulLeCam/react-leaflet#187]
Please make sure to check the following boxes before submitting an issue. Thanks!

- [x] Check that all peer dependencies are installed: React, ReactDOM and Leaflet.
- [x] Check that you are using a supported version of React (v0.14 or v15).
- [x] Check that you are using a supported version of Leaflet (v.0.7)
- [x] Make sure you have followed the [quick start guide](http://leafletjs.com/examples/quick-start.html) for Leaflet.
- [x] Make sure you have fully read the [documentation](https://github.com/PaulLeCam/react-leaflet/blob/master/README.md) and that you understand the [technical considerations](https://github.com/PaulLeCam/react-leaflet/blob/master/README.md#technical-considerations).

### Expected behavior
User navigates from map view to another view without issue.

### Actual behavior
When attempting to navigate away from a map with multiple base layers, I am seeing ```Uncaught TypeError: Cannot read property 'hasLayer' of null```.  I'm using Leaflet 0.7.7, React 15.1.0 and React-Leaflet 0.12.0.

Stack Trace:
```javascript
o.Control.Layers.o.Control.extend._addItem	@	leaflet.js:9
o.Control.Layers.o.Control.extend._update	@	leaflet.js:9
o.Control.Layers.o.Control.extend.removeLayer	@	leaflet.js:9
removeLayerControl	@	react-leaflet.js:3997
componentWillUnmount	@	react-leaflet.js:3860
```

If I only use one BaseLayer, everything behaves as expected.  The map works as expected until changing views.

This issue seems related to #145.

### Steps to reproduce
See [jsfiddle](https://jsfiddle.net/dkrichards86/dv0j9ac0/).

**Comment by chriswiggins** [comment]
I'm also seeing this exact same issue. I have two base layers and an overlay if thats at all helpful.

As per @dkrichards86 's stack trace above, L.Control.Layers tries to call addItem when it is removed, therefore causing issues. I can see why this is useful if dynamically removing/adding layers for the control, however a complete unmount doesn't work with this logic.

**Comment by PaulLeCam** [comment]
Hi,
Thanks for reporting this, indeed it seems `L.Control.Layers` tries to add layers to the map when it gets updated after being removed, but the map itself is removed as well so it breaks.
I pushed a fix in v0.12.1 that should address the issue, please let me know if it works in your case.

**Comment by chriswiggins** [comment]
You're amazing Paul. Will give this a go tomorrow.

Cheers
Chris

On 8/07/2016, at 8:55 PM, Paul Le Cam <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,
Thanks for reporting this, indeed it seems L.Control.Layers tries to add layers to the map when it gets updated after being removed, but the map itself is removed as well so it breaks.
I pushed a fix in v0.12.1 that should address the issue, please let me know if it works in your case.

-

**Comment by dkrichards86** [comment]
Thanks for the fix!  Everything is behaving as expected.
